### PR TITLE
Dashboard: Update Card Menu sizing to fit content better

### LINKED ISSUE #12559: Dashboard menu items cut off
**Describe the bug**
A clear and concise description of what the bug is. Please be as descriptive as possible; issues lacking detail, or for any other reason than to report a bug, may be closed without action.

If the dashboard card title is long, it will get cut off in the dismiss option menu. Make the menu option 2 lines if this happens. 

**To Reproduce**
1. Load the app dashboard and enable the "Most recent reviews" card. 
2. Switch app language to Spanish (this bug will likely happen in other languages, but in Spanish is were I first saw it)
3. Click on the 3 dots menu of the "Most recent reviews" card. 
4. See the item is cut off. 

**Screenshots**
Screenshot 2024-09-09 at 08 54 06
add screenshots to help explain your problem.

**Expected behavior**

Make the item 2 lines if the text is long. 

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active,
  or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: Pixel 8 
- Android version: 14
- WooCommerce Android version: 20.2

</details>

### PR DESCRIPTION
**Pull request by hafizrahman**
Closes: #12559

### Description

In #12559 it is reported that when the Card title is very long, the text gets cut off.

While testing with big fonts I also found that the menu generally doesn't fit the text correctly, due to it being set having an exact amount of height.

This PR updates the menu's height to be more flexible and fits bigger font sizes, as well as longer text.

### Steps to reproduce

1. Set simulator to biggest font size, then run app,
2. On Dashboard, enable as many cards as you can,
3. Check each card's menu item(s) and ensure they're shown correctly, and the text not cut off. Especially also for "Store Setup" that has two menu items.

### The tests that have been performed

The above

### Images/gif

Wrong states:
| Store setup | Long text |
|-|-|
| Screenshot_1734857191 | Screenshot_1734857205 |

Correct states:
| Store setup | Long text |
|-|-|
| Screenshot_1734857007 | Screenshot_1734856802 |

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/OverflowMenu.kt [hunk 1]
```diff
@@ -54,7 +55,7 @@ fun <T> WCOverflowMenu(
         ) {
             items.forEachIndexed { index, item ->
                 DropdownMenuItem(
-                    modifier = Modifier.height(dimensionResource(id = dimen.major_175)),
+                    modifier = Modifier.heightIn(min = dimensionResource(id = dimen.major_175)),
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/OverflowMenu.kt]: I wonder why we specify the size here. The [default sizes](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/material/material/src/commonMain/kotlin/androidx/compose/material/Menu.kt;l=243-248?q=DropdownMenuContent) for `DropdownMenuItem` already seem fine to me.

I removed this modifier line and tested it, and it resolves the mentioned issue while behaving the same as this PR. Could you also test it to check if I missed anything? If the behavior is the same, I’d prefer removing this modifier for simplicity.

Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/OverflowMenu.kt]: Nice investigation and good point! I didn't check too deeply and just wanted to preserve it, but I agree default is a better idea. I removed the part and tested it to work well in different cases. https://github.com/woocommerce/woocommerce-android/pull/13184/commits/96c0e5306fbe8f5b5b8e63a3dd5a59d0947069d0

**Review by irfano — APPROVED** [review]
LGTM! We're good to merge after fixing the detekt error.

**Pull request merged**